packages/liveblocks-react/src/liveblocks.tsx

```diff
@@ -223,31 +223,43 @@ export function getLiveblocksExtrasForClient<M extends BaseMetadata>(
 }
 
 function makeDeltaPoller_Notifications(store: UmbrellaStore<BaseMetadata>) {
-  const poller = makePoller(async () => {
-    try {
-      await store.waitUntilNotificationsLoaded();
-      await store.fetchNotificationsDeltaUpdate();
-    } catch (err) {
-      // When polling, we don't want to throw errors, ever
-      console.warn(`Polling new inbox notifications failed: ${String(err)}`);
-    }
-  }, POLLING_INTERVAL);
+  const poller = makePoller(
+    (signal) => {
+      // XXX - Spend some time thinking about the usage of AbortSignal here
+      void store.fetchNotificationsDeltaUpdate({ signal }).catch((err) => {
+        console.warn(`Polling new inbox notifications failed: ${String(err)}`);
+      });
+    },
+    60_000, // Polling interval is every 60 seconds
+    { maxStaleTimeMs: 5 * 1000 } // ...or if window refocuses and data is less than 5 seconds old
+  );
 
   // Keep track of the number of subscribers
   let pollerSubscribers = 0;
 
   return () => {
     pollerSubscribers++;
 
-    // NIMESH - We should wait until the lastRequestedAt date is known using a promise and then
+    // XXX - We should wait until the lastRequestedAt date is known using a promise and then
     // in the `then` body, check again if the number of subscribers if more than 0, and only then
     // if those conditions hold, start the poller
     poller.enable(pollerSubscribers > 0);
+    void store
+      .waitUntilNotificationsLoaded()
+      .then(() => {
+        poller.enable(pollerSubscribers > 0);
+      })
+      .catch((err) => {
+        // XXX - Think about a better message and if we need to handle this differently
+        console.warn(
+          `Unable to start poller due to failure loading first page of inbox notifications: ${String(err)}`
+        );
+      });
 
     return () => {
       pollerSubscribers--;
 
-      // NIMESH - When stopping the poller, we should also ideally abort its
+      // XXX - When stopping the poller, we should also ideally abort its
       // poller function, maybe using an AbortController? This functionality
       // should be automatic and handled by the Poller abstraction, not here!
```

Comment:
Having second thoughts about this comment. In the current implementation, when the machine is in `@polling` state, it won't respond to `STOP` events — by design. I think it makes sense to not abort the currently-ongoing poll and just handle its response, even if there are no subscribers anymore by the time the response arrives. This means the current implementation is already correct, and I think we can safely remove this comment.

Do you agree, @nimeshnayaju?